AstarteDeviceSDKCSharp/Device/AstarteDevice.cs

```diff
@@ -180,7 +180,19 @@ private bool EventualyReconnect()
                     {
                         interval = MAX_INCREMENT_INTERVAL;
                     }
-                    Task.Run(async () => await Connect()).Wait(interval);
+
+                    try
+                    {
+                        Task.Run(async () => await Connect()).Wait(interval);
+                    }
+                    catch (AggregateException ex)
```

Comment:
Why did you catch all exception with `AggregateException` here?

Reply:
If does not catch this exception, produced mess in threads. Do you have better solution for handling exception? https://github.com/dotnet/MQTTnet/blob/f79649d71b6544952986bac7f645482617a613ba/Samples/Client/Client_Connection_Samples.cs#L68